File: hackerrank/30Days/Day4.py
class Person:
    def __init__(self,initialAge):
        if(initialAge<0):
            print "Age is not valid, setting age to 0."
            initialAge = 0;
        self.age = initialAge;
    def amIOld(self):
        res = "young." if self.age<13 else "a teenager." if self.age>=13 and self.age<18 else "old."
        print "You are " + res
    def yearPasses(self):
        self.age+=1;
       
t = int(raw_input())
for i in range(0, t):
    age = int(raw_input())         
    p = Person(age)  
    p.amIOld()
    for j in range(0, 3):
        p.yearPasses()        
    p.amIOld()
    print("")    

File: hackerrank/30Days/Day4.java
/*Write a Person class with an instance variable, , and a constructor that takes an integer, , as a parameter. The constructor must assign  to  after confirming the argument passed as  is not negative; if a negative argument is passed as , the constructor should set  to  and print Age is not valid, setting age to 0.. In addition, you must write the following instance methods:

yearPasses() should increase the  instance variable by .
amIOld() should perform the following conditional actions:
If , print You are young..
If  and , print You are a teenager..
Otherwise, print You are old..
To help you learn by example and complete this challenge, much of the code is provided for you, but you'll be writing everything in the future. The code that creates each instance of your Person class is in the main method. Don't worry if you don't understand it all quite yet!*/

import java.io.*;
import java.util.*;

public class Person {
    private int age;	
  
	public Person(int initialAge) {
  		if(initialAge<0){
            System.out.println("Age is not valid, setting age to 0.");
            initialAge = 0;
        }
        this.age = initialAge;
	}

	public void amIOld() {
  		String res = " ";
        res = (this.age<13)?"young.":(this.age>=13 && this.age<18)?"a teenager.":"old.";
        System.out.println("You are " + res);
	}

	public void yearPasses() {
  		this.age++;
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		for (int i = 0; i < T; i++) {
			int age = sc.nextInt();
			Person p = new Person(age);
			p.amIOld();
			for (int j = 0; j < 3; j++) {
				p.yearPasses();
			}
			p.amIOld();
			System.out.println();
        }
		sc.close();
    }
}
